src/internal/mersenne/mersenne.ts

```
@@ -1,81 +1,64 @@
-   *
-   * @example
-   * faker.mersenne.rand() // 15515
-   * faker.mersenne.rand(1000, 500) // 578
-   *
-   * @since 5.5.0
    */
-  rand(max = 32768, min = 0): number {
-    if (min > max) {
-      const temp = min;
-      min = max;
-      max = temp;
-    }
-
-    return Math.floor(this.gen.genrandReal2() * (max - min) + min);
-  }
+  next(max?: number, min?: number): number;
 
   /**
    * Sets the seed to use.
    *
-   * @param S The seed to use.
-   * @throws If the seed is not a `number`.
-   *
-   * @since 5.5.0
+   * @param seed The seed to use.
+   * @throws If the seed is not a `number` or `number[]`.
    */
-  seed(S: number): void {
-    if (typeof S !== 'number') {
-      throw new FakerError(
-        `seed(S) must take numeric argument; is ${typeof S}`
-      );
-    }
+  seed(seed: number | number[]): void;
+}
 
-    this.gen.initGenrand(S);
-  }
+/**
+ * Generate seed based random numbers.
+ *
+ * @internal
+ */
+export default function mersenne(): Mersenne {
+  const twister = new Twister();
+
+  twister.initGenrand(Math.ceil(Math.random() * Number.MAX_SAFE_INTEGER));
+
+  return {
+    next(max = 32768, min = 0): number {
+      if (min > max) {
+        const temp = min;
+        min = max;
+        max = temp;
+      }
+
+      return Math.floor(twister.genrandReal2() * (max - min) + min);
+    },
+
+    seed(seed: number | number[]): void {
+      if (typeof seed === 'number') {
+        twister.initGenrand(seed);
+        return;
+      }
+
+      if (Array.isArray(seed)) {
+        twister.initByArray(seed, seed.length);
+        return;
+      }
 
-  /**
-   * Sets the seed to use.
-   *
-   * @param A The seed to use.
-   * @throws If the seed is not a `number[]`.
-   *
-   * @since 5.5.0
-   */
-  seed_array(A: number[]): void {
-    if (typeof A !== 'object') {
       throw new FakerError(
-        `seed_array(A) must take array of numbers; is ${typeof A}`
+        `seed must take numeric argument(s); is ${typeof seed}`
       );
```

Do we really need to throw an error here, as the function will return void anyway?

---

Same as @ST-DDT wrote: because it's now internal and we always new what the input is, I also agree we don't need this check anymore. At least not here, but maybe in faker.seed 🤔

---

We usually dont handle badly typed input so I assume we dont need it at all.

---

removed